test/WebJobs.Script.Tests.Integration/Host/StandbyManagerTests.cs

```diff
@@ -130,7 +137,123 @@ public void IsWarmUpRequest_ReturnsExpectedValue()
                 await TestHelpers.Await(() =>
                 {
                     // wait for the trace indicating that the host has been specialized
-                    logLines = loggerProvider.GetAllLogMessages().Select(p => p.FormattedMessage).ToArray();
+                    logLines = loggerProvider.GetAllLogMessages().Where(p => p.FormattedMessage != null).Select(p => p.FormattedMessage).ToArray();
+                    return logLines.Contains("Generating 0 job function(s)");
+                }, userMessageCallback: () => string.Join(Environment.NewLine, loggerProvider.GetAllLogMessages().Select(p => $"[{p.Timestamp.ToString("HH:mm:ss.fff")}] {p.FormattedMessage}")));
+
+                httpServer.Dispose();
+                httpClient.Dispose();
+
+                await Task.Delay(2000);
+
+                var hostConfig = WebHostResolver.CreateScriptHostConfiguration(webHostSettings, true);
+                var expectedHostId = hostConfig.HostConfig.HostId;
+
+                // verify the rest of the expected logs
+                string text = string.Join(Environment.NewLine, logLines);
+                Assert.True(logLines.Count(p => p.Contains("Stopping Host")) >= 1);
+                Assert.Equal(1, logLines.Count(p => p.Contains("Creating StandbyMode placeholder function directory")));
+                Assert.Equal(1, logLines.Count(p => p.Contains("StandbyMode placeholder function directory created")));
+                Assert.Equal(2, logLines.Count(p => p.Contains("Starting Host (HostId=placeholder-host")));
+                Assert.Equal(2, logLines.Count(p => p.Contains("Host is in standby mode")));
+                Assert.Equal(2, logLines.Count(p => p.Contains("Executed 'Functions.WarmUp' (Succeeded")));
+                Assert.Equal(1, logLines.Count(p => p.Contains("Starting host specialization")));
+                Assert.Equal(1, logLines.Count(p => p.Contains($"Starting Host (HostId={expectedHostId}")));
+                Assert.Contains("Generating 0 job function(s)", logLines);
+
+                WebScriptHostManager.ResetStandbyMode();
+            }
+        }
+
+        [Fact]
+        public async Task StandbyMode_EndToEnd_LinuxContainer()
+        {
+            byte[] bytes = TestHelpers.GenerateKeyBytes();
+            var encryptionKey = Convert.ToBase64String(bytes);
+
+            var vars = new Dictionary<string, string>
+            {
+                { EnvironmentSettingNames.ContainerName, "TestContainer" },
```

Comment:
@ahmelsayed Do these pre-specialization environment variables look right to you?

Reply:
Yep, they look right to me. The only one I'm not familiar with is `AzureWebEncryptionKey`, not sure if that's something that's used in Fabio's data protection lib?